### ISSUE TITLE
Unable to install

### ISSUE THREAD

**Issue by rafaelrinaldi** [dj95/zjstatus#72]
**Describe the bug**
I am unable to install the plugin.

With an empty configuration file, if I just add the default layout file as suggested in the readme, I am unable to open Zellij and see a "Received empty message from server" message.

**To Reproduce**
1. Empty Zellij config to ensure nothing is interfering with plugin installation
2. Copy and paste layout code into `layouts/default.kdl`
3. Run `zellij -s test` in the terminal
4. See error: "Received empty message from server" message

**Expected behavior**
Zellij opens normally and I am able to see a status bar rendered by zjstatus.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS Sonoma 14.5
 - Zellij version v0.40.1

**Layout**

How does the layout look like? Please copy it into a code block.

```kdl
layout {
  default_tab_template {
    children
    pane size=1 borderless=true {
      plugin location="https://github.com/dj95/zjstatus/releases/latest/download/zjstatus.wasm" {
        format_left   "{mode} #[fg=#89B4FA,bold]{session}"
        format_center "{tabs}"
        format_right  "{command_git_branch} {datetime}"
        format_space  ""

        border_enabled  "false"
        border_char     "─"
        border_format   "#[fg=#6C7086]{char}"
        border_position "top"

        hide_frame_for_single_pane "true"

        mode_normal  "#[bg=blue] "
        mode_tmux    "#[bg=#ffc387] "

        tab_normal   "#[fg=#6C7086] {name} "
        tab_active   "#[fg=#9399B2,bold,italic] {name} "

        command_git_branch_command     "git rev-parse --abbrev-ref HEAD"
        command_git_branch_format      "#[fg=blue] {stdout} "
        command_git_branch_interval    "10"
        command_git_branch_rendermode  "static"

        datetime        "#[fg=#6C7086,bold] {format} "
        datetime_format "%A, %d %b %Y %H:%M"
        datetime_timezone "Europe/Berlin"
      }
    }
  }
}
```

**Additional context**
N/A

**Comment by dj95** [comment]
Hi and thanks for the bug report.

Can you please try to load zjstatus via file? On macOS the https locations sometimes seem to be buggy.

**Comment by PatschD** [comment]
get the same error on macos, even when i load a file

**Comment by dj95** [comment]
Have you also tried to clean the cache? What does the logs of zellij say?

**Comment by PatschD** [comment]
do you know where the logs are located on macos? i cleared cache and that didnt help

**Comment by dj95** [comment]
You'll find the details here: https://zellij.dev/documentation/plugin-api-logging.html?highlight=logs#plugin-api---logging

Which version of zjstatus do you use? Unfortunately I cannot reproduce the issue on 2 MacBooks.

**Comment by PatschD** [comment]
ERROR  |zellij_server::plugins   | 2024-06-21 14:50:47.736 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:51:48.083 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:52:48.415 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:53:48.752 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:54:49.061 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:55:49.376 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:56:49.684 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:57:49.969 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:58:50.280 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 14:59:50.605 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 15:00:50.927 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 15:01:51.242 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 15:02:51.538 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 15:03:51.817 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
ERROR  |zellij_server::plugins   | 2024-06-21 15:04:52.095 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found 
INFO   |zellij_client            | 2024-06-21 15:05:10.737 [main      ] [zellij-client/src/lib.rs:178]: Starting Zellij client! 
INFO   |zellij_server            | 2024-06-21 15:05:10.751 [main      ] [zellij-server/src/lib.rs:284]: Starting Zellij server! 
ERROR  |zellij_client::stdin_ansi| 2024-06-21 15:05:10.809 [stdin_handler] [zellij-client/src/stdin_ansi_parser.rs:124]: Failed to open STDIN cache file: Os { code: 2, kind: NotFound, message: "No such file or directory" } 
INFO   |zellij_server            | 2024-06-21 15:05:10.808 [main      ] [zellij-server/src/lib.rs:1068]: Compiling plugins using Cranelift 
INFO   |zellij_server::plugins   | 2024-06-21 15:05:10.809 [wasm      ] [zellij-server/src/plugins/mod.rs:203]: Wasm main thread starts 
WARN   |zellij_utils::ipc        | 2024-06-21 15:05:10.832 [router    ] [zellij-utils/src/ipc.rs:226]: Error in IpcReceiver.recv(): InvalidMarkerRead(Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }) 
ERROR  |zellij_client            | 2024-06-21 15:05:10.832 [router    ] [zellij-client/src/lib.rs:375]: Received empty message from server 
ERROR  |zellij_server::plugins   | 2024-06-21 15:05:52.428 [wasm      ] [zellij-server/src/plugins/mod.rs:784]: Plugin with id: 0 not found

**Comment by dj95** [comment]
After you've cleared the cache, does the permission request pop up? Does the error occur before the permissions or after you've granted them? 

Which zjstatus version are you using?

**Comment by PatschD** [comment]
<img width="632" alt="image" src="https://github.com/dj95/zjstatus/assets/48880710/f0c23d27-d7cb-4bf6-9bb0-3a30b6d39200">
<img width="935" alt="image" src="https://github.com/dj95/zjstatus/assets/48880710/b808da96-b8f5-4713-a106-fda58dbde6a0">

layout {
    default_tab_template {
        children
        pane size=1 borderless=true {
            plugin location="https://github.com/dj95/zjstatus/releases/latest/download/zjstatus.wasm" {
                format_left  "#[fg=0,bg=10][{session}]  {tabs}"
                format_right "#[fg=0,bg=10]{mode}"
                format_space "#[bg=10]"

                mode_normal        "{name} "
                mode_tmux          "{name} "
                mode_locked        "{name} "
                mode_resize        "{name} "
                mode_pane          "{name} "
                mode_tab           "{name} "
                mode_scroll        "{name} "
                mode_enter_search  "{name} "
                mode_search        "{name} "
                mode_rename_tab    "{name} "
                mode_rename_pane   "{name} "
                mode_session       "{name} "
                mode_move          "{name} "

                tab_normal   "{index}:{name}  "
                tab_active   "{index}:{name}* "
            }
        }
    }
}

**Comment by PatschD** [comment]
no permission request just error in plugin

**Comment by PatschD** [comment]
indeed this works! thanks.
i initially tried it with local file but i added an additional dot.
i copied from your docs: 
<img width="826" alt="image" src="https://github.com/dj95/zjstatus/assets/48880710/4967aaca-8750-480b-be85-a699596cf741">

as 
```
file:./path
```
rather than
```
file:/path
```

**Comment by dj95** [comment]
Switched it now to `~/path/`, such that it will be able to load the file relative to the home directory. Relative path's also work, but I guess only when you open zellij near the layout kdl file.

**Comment by dj95** [comment]
Found another issue within zellij, if multiple tabs are loaded in parallel without a cache zjstatus version.

@rafaelrinaldi It would be great if you can confirm that loading the plugin via file works.

**Comment by dj95** [comment]
As there's no response, I assume, that there is no further issue, when setting it via the `file:` protocol and will close the issue. 

Feel free to add any details in case something is missing.

**Comment by rafaelrinaldi** [comment]
@dj95 I was finally able to revisit this and it still doesn't work. I have manually downloaded the `wasm` file and moved it to `~/dev/zjstatus.wasm` with this config (under `~/.config/zellij/layouts/default.kdl`):

```kdl
layout {
  pane split_direction="vertical" {
    pane
  }

  pane size=1 borderless=true {
    plugin location="file:~/dev/zjstatus.wasm" {
      hide_frame_for_single_pane "true"

      format_left  "{mode}#[fg=#89B4FA,bg=#181825,bold] {session}#[bg=#181825] {tabs}"
      format_right "{command_kubectx}#[fg=#424554,bg=#181825]::{command_kubens}{datetime}"
      format_space "#[bg=#181825]"

      mode_normal          "#[bg=#89B4FA] "
      mode_tmux            "#[bg=#ffc387] "
      mode_default_to_mode "tmux"

      tab_normal               "#[fg=#6C7086,bg=#181825] {index} {name} {fullscreen_indicator}{sync_indicator}{floating_indicator}"
      tab_active               "#[fg=#9399B2,bg=#181825,bold,italic] {index} {name} {fullscreen_indicator}{sync_indicator}{floating_indicator}"
      tab_fullscreen_indicator "□ "
      tab_sync_indicator       "  "
      tab_floating_indicator   "󰉈 "

      command_kubectx_command  "kubectx -c"
      command_kubectx_format   "#[fg=#6C7086,bg=#181825,italic] {stdout}"
      command_kubectx_interval "2"

      command_kubens_command  "kubens -c"
      command_kubens_format   "#[fg=#6C7086,bg=#181825]{stdout} "
      command_kubens_interval "2"

      datetime          "#[fg=#9399B2,bg=#181825] {format} "
      datetime_format   "%A, %d %b %Y %H:%M"
      datetime_timezone "Europe/Berlin"
    }
  }
}
```

Zellij renders but I see an error message at the bottom that says:

>ERROR IN PLUGIN - check logs for more info

I can't seem to find where these logs are being stored. They're not being piped to stderr.

**Comment by dj95** [comment]
Hi @rafaelrinaldi 

Can you see something in zellij log? It's somewhere hidden in /var.

Does the error on your side pop up before or after granting permissions to zjstatus?

**Comment by philocalyst** [comment]
Seeing something similar on my end, where the statusbar isn't loading.

<img width="1512" alt="Image" src="https://github.com/user-attachments/assets/3e9678b8-a25d-41be-887e-f42d098dd103" />

That's using this layout:
```kdl
layout {

    pane size=1 borderless=true {

        plugin location="file:~/.config/zellij/Plugins/zjstatus.wasm" {

            hide_frame_for_single_pane "true"


            format_left  "{mode}#[fg=#89B4FA,bg=#181825,bold] {session}#[bg=#181825] {tabs}"

            format_right "{command_kubectx}#[fg=#424554,bg=#181825]::{command_directory}{datetime}"

            format_space "#[bg=#181825]"



            mode_normal          "#[bg=#89B4FA] "

            mode_locked          "#[bg=#ffc387] "

            mode_default_to_mode "locked"


            tab_active               "#[fg=#FFFFFF,bg=#181825]|{name}| {fullscreen_indicator}{sync_indicator}{floating_indicator}"

            tab_normal              "#[fg=#6399B2,bg=#181825,bold,italic]|{name}| {fullscreen_indicator}{sync_indicator}{floating_indicator}"

            tab_fullscreen_indicator "□ "

            tab_sync_indicator       "  "

            tab_floating_indicator   "󰉈 "


            command_directory_command  "bash -c \"pwd | xargs basename\""

            command_directory_format   "#[fg=#6C7086,bg=#181825] {stdout}"

            command_directory_interval "1"


            datetime          "#[fg=#9399B2,bg=#181825] {format} "

            datetime_format   "%H:%M"

            datetime_timezone "EST"

        }

    }

    pane

} 
```

And my logfile is such:
```md
INFO   |zellij_client            | 2025-02-23 10:26:26.679 [main      ] [zellij-client/src/lib.rs:182]: Starting Zellij client! 
INFO   |zellij_server            | 2025-02-23 10:26:26.690 [main      ] [zellij-server/src/lib.rs:522]: Starting Zellij server! 
WARN   |zellij_utils::kdl        | 2025-02-23 10:26:26.693 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file 
WARN   |zellij_utils::kdl        | 2025-02-23 10:26:26.693 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file 
WARN   |zellij_utils::kdl        | 2025-02-23 10:26:26.693 [main      ] [zellij-utils/src/kdl/mod.rs:695]: Converting new tab action without arguments, original action saved to .bak.kdl file 
ERROR  |zellij_client::stdin_ansi| 2025-02-23 10:26:26.694 [stdin_handler] [zellij-client/src/stdin_ansi_parser.rs:124]: Failed to open STDIN cache file: Os { code: 2, kind: NotFound, message: "No such file or directory" } 
INFO   |zellij_server            | 2025-02-23 10:26:26.694 [main      ] [zellij-server/src/lib.rs:1493]: Compiling plugins using Cranelift 
INFO   |zellij_server::plugins   | 2025-02-23 10:26:26.694 [wasm      ] [zellij-server/src/plugins/mod.rs:232]: Wasm main thread starts 
INFO   |zellij_server::plugins::p| 2025-02-23 10:26:26.699 [async-std/runti] [zellij-server/src/plugins/plugin_loader.rs:533]: Loaded plugin '/Users/philocalyst/.config/zellij/Plugins/zjstatus.wasm' from cache folder at '/Users/philocalyst/Library/Caches/org.Zellij-Contributors.Zellij' in 704.625µs 
ERROR  |zellij_utils::input::perm| 2025-02-23 10:26:26.700 [async-std/runti] [zellij-utils/src/input/permission.rs:51]: Failed to read permission cache file: No such file or directory (os error 2) 
```
